src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift

```diff
@@ -0,0 +1,411 @@
+            openButton.tintColor = UIColor.buttonTint
+            view?.addSubview(openButton)
+        }
+    }
+
+    func addDownloadButton(to view: UIView?, withTarget target: Any?) {
+        let xPos = (view?.frame.size.width ?? 0.0) - 75
+        let yPos = (view?.frame.size.height ?? 0.0) * 0.1 + Util.screenHeight() / 4.5
+        let downloadButton = RoundBorderedButton(frame: CGRect(x: xPos, y: yPos - 25, width: 70, height: 25), andBorder: true)
+
+        if let downloadButton = downloadButton {
+            downloadButton.tag = Int(kDownloadButtonTag)
+            downloadButton.titleLabel?.font = UIFont.boldSystemFont(ofSize: 14)
+            downloadButton.titleLabel?.adjustsFontSizeToFitWidth = true
+            downloadButton.titleLabel?.minimumScaleFactor = 0.4
+            downloadButton.setTitle(kLocalizedDownload, for: .normal)
+            downloadButton.tintColor = UIColor.buttonTint
+
+            downloadButton.addTarget(target, action: #selector(downloadButtonPressed), for: .touchUpInside)
+
+            let activity = UIActivityIndicatorView(style: .gray)
+            activity.tag = Int(kActivityIndicator)
+            activity.frame = CGRect(x: 5, y: 0, width: 25, height: 25)
+            downloadButton.addSubview(activity)
+
+            view?.addSubview(downloadButton)
+        }
+    }
+
+    func addThumbnailImage(withImageUrlString imageUrlString: String?, to view: UIView?) {
+        let imageView = UIImageView()
+        let errorImage = UIImage(named: "thumbnail_large")
+
+        imageView.frame = CGRect(x: (view?.frame.size.width ?? 0.0) / 15, y: (view?.frame.size.height ?? 0.0) * 0.1, width: (view?.frame.size.width ?? 0.0) / 3, height: Util.screenHeight() / 4.5)
+        imageView.image = UIImage(
+            contentsOf: URL(string: imageUrlString ?? ""),
+            placeholderImage: nil,
+            errorImage: errorImage,
+            onCompletion: { image in
+                DispatchQueue.main.async(execute: {
+                    imageView.viewWithTag(Int(kActivityIndicator))?.removeFromSuperview()
+                    imageView.image = image
+                })
+            })
+
+        if imageView.image == nil {
+            let activity = UIActivityIndicatorView(style: .gray)
+            activity.tag = Int(kActivityIndicator)
+            activity.frame = CGRect(x: imageView.frame.size.width / 2.0 - 25.0 / 2.0, y: imageView.frame.size.height / 2.0 - 25.0 / 2.0, width: 25.0, height: 25.0)
+            imageView.addSubview(activity)
+            activity.startAnimating()
+        }
+
+        imageView.contentMode = .scaleAspectFill
+        imageView.layer.cornerRadius = 8.0
+        imageView.layer.masksToBounds = true
+        imageView.layer.borderColor = UIColor.utilityTint.cgColor
+        imageView.layer.borderWidth = 1.0
+
+        view?.addSubview(imageView)
+    }
+
+    func addProjectDescriptionLabel(withDescription description: String?, to view: UIView?, target: Any?) {
```

Comment:
`description` and `view` should not be optional. Also set scope to `fileprivate`.